runtime/lua/vim/lsp.lua

```diff
@@ -488,14 +586,39 @@ function lsp.start_client(config)
     end)
   end
 
+  --@private
+  --- Throws error for a method that is not supported by the current LSP
+  --- server.
+  ---
+  --@param method (string) an LSP method name not supported by the LSP server.
+  --@returns (error) a 'MethodNotFound' JSON-RPC error response.
   local function unsupported_method(method)
     local msg = "server doesn't support "..method
     local _ = log.warn() and log.warn(msg)
     err_message(msg)
     return lsp.rpc_response_error(protocol.ErrorCodes.MethodNotFound, msg)
   end
 
-  --- Checks capabilities before rpc.request-ing.
+  --@private
```

Comment:
I'd have to check again whether this is actually the case, but can't users access all of the `client.*` style functions from the returned client object?

Reply:
Yeah, AFAIK they can. (Are you asking why I marked all the `client.` methods as private? These methods are exposed in the vim docs with the `client.` prefix stripped off, and  I couldn't get lua2dox/gen_vimdoc.py to change that. So I just marked them as private and decided to document them on the client() function instead.)

Reply:
The reason why I documented the `client.` functions in two locations is that Lua langservers take their docs from the function docstrings, while the vimdocs are generated from the `client()` object, so I'll have to keep both in sync (for now).

Reply:
Hmmm OK. Maybe just make a note of that (for future readers)

Reply:
Will do.

Is there anything else I should change about the functions on the client object? That's definitely the part I was the most unsure about.

Reply:
Added a note and cleaned up the docs on the client object somewhat.